backend/src/layer/data/flickr_api/json_parser.rs

```diff
@@ -55,17 +55,17 @@ impl JsonParser {
     /// A boolean if the image has an valid license or not.
     /// If the image has no license or no license history, the image isn't restricted by any license and true will be returned.
     #[must_use]
-    pub fn check_license(root: &JsonRootLicense) -> bool {
+    pub fn check_license(root: &JsonRootLicense) -> (bool, String) {
         let license = root
             .license_history
             .iter()
             .max_by_key(|l| l.date_change)
             .map(|entry| &entry.new_license);
 
         if let Some(license) = license {
-            return VALID_LICENSES.contains(&license.as_str());
+            return (VALID_LICENSES.contains(&license.as_str()), license.clone());
```

Comment:
no need to clone.
if the &license for .contains makes problems, you could just put that into an extra line before.